src/datasets/utils/py_utils.py

```diff
@@ -325,12 +325,41 @@ def map_nested(
     map_tuple: bool = False,
     map_numpy: bool = False,
     num_proc: Optional[int] = None,
-    types=None,
+    types: Optional[tuple] = None,
     disable_tqdm: bool = True,
     desc: Optional[str] = None,
 ):
     """Apply a function recursively to each element of a nested data struct.
-    If num_proc > 1 and the length of data_struct is longer than num_proc: use multi-processing
+
+    Use multiprocessing if num_proc > 1 and the length of data_struct is larger than 1.
+
+    <Changed version="2.4.0">
+
+    Before version 2.4.0, multiprocessing was not used if `num_proc` was greater than or equal to ``len(iterable)``.
+
+    Now, if `num_proc` is greater than or equal to ``len(iterable)``, `num_proc` is set to ``len(iterable)`` and
+    multiprocessing is used.
+
+    </Changed>
+
+    Args:
+        function (Callable): Function to be applied to `data_struct`.
+        data_struct: Data structure to apply `function` to.
+        dict_only (bool, default False): Whether only apply `function` recursively to `dict` values in `data_struct`.
```

Comment:
Nit - Maybe something like this for the args listed here? Not sure we have a standard across the package.

```suggestion
        dict_only (`bool`, default `False`): Whether only apply `function` recursively to `dict` values in `data_struct`.
```

Otherwise I would put "Defaults to `False` a the end of the description itself and leave the annotation as (`bool`).

Reply:
Definitely, you are right. I didn't pay too much attention to the syntax because this docstring is not published in our docs. But I agree it is better to do it right in case it will be published some day. I'm fixing it.